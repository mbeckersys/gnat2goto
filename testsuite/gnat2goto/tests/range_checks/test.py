from test_support import *

# the following needs merge from branch feature/loops
#prove("--show-goto-functions")

prove()
