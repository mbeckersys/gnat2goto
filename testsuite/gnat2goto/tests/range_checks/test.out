CBMC version 5.6 64-bit x86_64 linux
Parsing range_checks.json_symtab
Converting
Generating GOTO Program
Adding CPROVER library (x86_64)
Removal of function pointers and virtual functions
Partial Inlining
Generic Property Instrumentation
Starting Bounded Model Checking
size of program expression: 34 steps
simple slicing removed 5 assignments
Generated 10 VCC(s), 10 remaining after simplification
Passing problem to propositional reduction
converting SSA
Running propositional reduction
Post-processing
Solving with MiniSAT 2.2.1 with simplifier
193 variables, 0 clauses
SAT checker: instance is SATISFIABLE
Solving with MiniSAT 2.2.1 with simplifier
193 variables, 0 clauses
SAT checker inconsistent: instance is UNSATISFIABLE
Runtime decision procedure: 0.007s

** Results:
[overflow.1] overflow on bounded type conversion in (int)range_checks__add__b: FAILURE
[overflow.2] overflow on bounded type conversion in (int)(range_checks__add__a + (int)range_checks__add__b): FAILURE
[overflow.3] overflow on bounded type conversion in (int)range_checks__c: SUCCESS
[overflow.4] overflow on bounded type conversion in (int)(range_checks__add__a + range_checks__add__b + (int)range_checks__c): FAILURE
[overflow.5] overflow on bounded type conversion in (int)(range_checks__c + range_checks__c + (int)range_checks__c): FAILURE
[overflow.6] overflow on bounded type conversion in (int)-5: FAILURE
[overflow.7] overflow on bounded type conversion in (int)range_checks__n1: SUCCESS
[overflow.8] overflow on bounded type conversion in (int)(range_checks__n1 + range_checks__n1 + (int)range_checks__n1): FAILURE
[overflow.9] overflow on bounded type conversion in (int)1: SUCCESS
[overflow.10] overflow on bounded type conversion in (int)(range_checks__n2 + (int)1): FAILURE

** 7 of 10 failed (2 iterations)
VERIFICATION FAILED
